ODIN_II/SRC/include/SubcktTypes.h

```diff
@@ -0,0 +1,272 @@
+#ifndef __SUBCKT_TYPES_H__
+#define __SUBCKT_TYPES_H__
+/**
+ * Copyright (c) 2021 Seyed Alireza Damghani (sdamghann@gmail.com)
+ * 
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "odin_types.h"
+
+#define NUM_OF_ODIN_CELL_TYPES 4
+#define NUM_OF_YOSYS_CELL_TYPES 218
+
+// global hashmap of yosys subckt types variable
+typemap odin_subckt_str({{"multiply", MULTIPLY},
+                         {"mult_", MULTIPLY},
+                         {"adder", ADD},
+                         {"sub", MINUS}},
+                        NUM_OF_ODIN_CELL_TYPES);
+
+// global hashmap of yosys subckt types variable
+typemap yosys_subckt_str({
```

Comment:
Explain what these are.  What is the first entry in each pair, and what is the second?

Reply:
I actually moved these definitions to the` enum_str.cpp,` you would find more comments there